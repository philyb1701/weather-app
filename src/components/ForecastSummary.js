import React from "react";
import PropTypes from "prop-types";

function ForecastSummary(props) {
  const { date, temperature, description, icon } = props;
  return (
    <div className="forecast-summary">
      <div className="forecast-summary__date" data-testid="forecast-summary">
        {date}
      </div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        {icon}
      </div>
      <div className="forecast-summary__temp" data-testid="forecast-temp">
        {temperature.max}&deg;C
      </div>
      <div className="forecast-summary__desc" data-testid="forecast-desc">
        {description}
      </div>
    </div>
  );
}

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};
export default ForecastSummary;
