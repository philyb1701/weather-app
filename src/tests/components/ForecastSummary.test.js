import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 11111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders data correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(getByText("11111111")).toHaveClass("forecast-summary__date");
    expect(getByText("Stub description")).toHaveClass("forecast-summary__desc");
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getByText("22Â°C")).toHaveClass("forecast-summary__temp");
  });
});

// You can use the toHaveAttribute() method to check for the existence of any html attribute.

// In this example we checked for the class attribute to illustrate how to use the toHaveAttribute() method (once our code is compiled the react specific className attribute will become class). Though you'd typically want to avoid testing implementation details such as class names, more info here.
