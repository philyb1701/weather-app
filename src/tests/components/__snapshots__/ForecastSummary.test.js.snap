// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders data correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary__date"
      data-testid="forecast-summary"
    >
      11111111
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      stubIcon
    </div>
    <div
      class="forecast-summary__temp"
      data-testid="forecast-temp"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary__desc"
      data-testid="forecast-desc"
    >
      Stub description
    </div>
  </div>
</DocumentFragment>
`;
